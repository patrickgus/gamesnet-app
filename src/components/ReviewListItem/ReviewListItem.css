.ReviewListItem {
  margin: 0;
  padding: 10px 20px;
  background-color: #bbb;
}

.ReviewListItem:nth-child(even) {
  background-color: #ddd;
}
