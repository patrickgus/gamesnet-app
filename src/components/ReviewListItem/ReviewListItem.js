import React, { Component } from "react";
import { Section } from "../../components/Utils/Utils";
import "./ReviewListItem.css";

export default class ReviewListItem extends Component {
  render() {
    const { review } = this.props;
    return (
      <Section className="ReviewListItem">
        <h4>"{review.title}"</h4>
        <p>Rating: {review.rating}</p>
        <div className="reviewInfo">
          <p>{review.description}</p>
        </div>
      </Section>
    );
  }
}
