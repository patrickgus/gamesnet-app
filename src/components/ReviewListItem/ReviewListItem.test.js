import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import ReviewListItem from "./ReviewListItem";
import review from "../../stores/review-store";

describe("ReviewListItem component", () => {
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<ReviewListItem review={review} />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it("renders the item as expected", () => {
    const tree = renderer.create(<ReviewListItem review={review} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
