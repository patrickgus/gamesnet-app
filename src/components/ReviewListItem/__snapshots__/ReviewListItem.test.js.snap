// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReviewListItem component renders the item as expected 1`] = `
<section
  className="Section ReviewListItem"
>
  <h4>
    "
    "
  </h4>
  <p>
    Rating: 
  </p>
  <div
    className="reviewInfo"
  >
    <p />
  </div>
</section>
`;
